lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -128,29 +128,37 @@ export const migrateGranuleAndFilesViaTransaction = async (
   const collectionId = dynamoRecord.collectionId;
   const { granulesSummary, filesSummary } = granuleAndFileMigrationSummary;
 
+  granulesSummary.dynamoRecords += 1;
+  filesSummary.dynamoRecords += files.length;
+
   try {
-    granulesSummary.dynamoRecords += 1;
-    await migrateGranuleRecord(dynamoRecord, knex);
+    await knex.transaction(async (trx) => {
+      // Have to pass in the transaction (`trx`) so that it is used
+      // by the granule/files writes
+      await migrateGranuleRecord(dynamoRecord, trx);
+      await Promise.all(files.map(async (file : ApiFile) => {
+        try {
+          await migrateFileRecord(file, granuleId, collectionId, trx);
```

Comment:
Ah, so that was the key part I didn't understand (aka passing in the transaction to be used)